testsuite/modern-testsuite/src/test/java/org/sonatype/nexus/testsuite/maven/concurrency/MavenMetadataHostedSwarmIT.java

```diff
@@ -0,0 +1,147 @@
+
+    // create clients
+    final List<Callable<HttpResponse>> clients = new ArrayList<>();
+    for (int i = 0; i < 10; i++) {
+      clients
+          .add(new UriPut(repositoryClient, path, generator, Suppliers.ofInstance(ByteSize.bytes(100L)) /* unused */));
+    }
+    for (int i = 0; i < 15; i++) {
+      clients.add(new UriGet(repositoryClient, path));
+    }
+    for (int i = 0; i < 2; i++) {
+      clients.add(new Repeat(3, new RebuildMavenMetadata(repository.facet(MavenHostedFacet.class))));
+    }
+
+    // put initial as we don't know will GET or PUT "win"
+    final HttpEntity entity = generatedEntity(generator, ByteSize.bytes(100L) /* unused */);
+    HttpResponse response = repositoryClient.put(path, entity);
+    assertThat(response.getStatusLine().getStatusCode(), equalTo(201));
+
+    // except last two clients, that does not return HttpResponse code
+    assertAllHttpResponseIs2xx(performSwarm(clients).subList(0, clients.size() - 2));
+  }
+
+  @Test
+  public void groupLevel() throws Exception {
+    Metadata metadata = new Metadata();
+    metadata.setModelVersion("1.1.0");
+    metadata.setGroupId("org.sonatype.nexus.test");
+    Plugin plugin = new Plugin();
+    plugin.setPrefix("prefix");
+    plugin.setArtifactId("artifact");
+    plugin.setName("Name");
+    metadata.addPlugin(plugin);
+    hostedMetadataSwarm("org/sonatype/nexus/test/maven-metadata.xml", new MavenMetadataGenerator(metadata));
+  }
+
+  @Test
+  public void artifactLevel() throws Exception {
+    Metadata metadata = new Metadata();
+    metadata.setModelVersion("1.1.0");
+    metadata.setGroupId("org.sonatype.nexus.test");
+    metadata.setArtifactId("test-artifact");
+    Versioning versioning = new Versioning();
+    versioning.setLatest("1.0.0");
+    versioning.setRelease("1.0.0");
+    versioning.addVersion("0.0.8");
+    versioning.addVersion("0.0.9");
+    versioning.setLastUpdated("20150715120000");
+    metadata.setVersioning(versioning);
+    hostedMetadataSwarm("org/sonatype/nexus/test/test-artifact/maven-metadata.xml", new MavenMetadataGenerator(metadata));
+  }
+
+  @Test
+  public void versionLevel() throws Exception {
+    Metadata metadata = new Metadata();
+    metadata.setModelVersion("1.1.0");
+    metadata.setGroupId("org.sonatype.nexus.test");
+    metadata.setArtifactId("test-artifact");
+    metadata.setVersion("1.0-SNAPSHOT");
+    Versioning versioning = new Versioning();
+    Snapshot snapshot = new Snapshot();
+    snapshot.setTimestamp("20150715.120000");
+    snapshot.setBuildNumber(1);
+    versioning.setSnapshot(snapshot);
+    final List<SnapshotVersion> snapshotVersions = new ArrayList<>();
+    SnapshotVersion snapshotVersion = new SnapshotVersion();
+    snapshotVersion.setExtension("pom");
+    snapshotVersion.setUpdated("20150715120000");
+    snapshotVersion.setVersion("1.0-20150715.120000-1");
+    snapshotVersions.add(snapshotVersion);
+    snapshotVersion = new SnapshotVersion();
```

Comment:
Reuse of the variable name makes this harder to read, and potentially error-prone if we need to change it in the future
Also, just seems "wrong" that all three of these snapshots are exactly the same, although likely it has no impact on the testing